### DISCUSSION TITLE
Question. How to use it with helix graphql.

### DISCUSSION THREAD

**Discussion by blackpr** [enisdenjo/graphql-sse#10] [Q&A] [answered]
Hello,
If this is not the place for this question please feel free to close/delete it.

So. I have a graphq-helix graphql server.
I can connect to subscriptions with the buil-in graphiql
```
subscription BasicSubscription {
  clock
}
```

I can connect with `EventSource`
```
const eventSource = new EventSource(u)
      const payload = await new Promise<any>((resolve) => {
        eventSource.addEventListener('message', (event: any) => {
          console.log(event.data) // <------- this logs the message
          resolve(event.data)
          eventSource.close()
        })
      })
```

But I cannot connect with this client nor apollo. I get the same error `Error: Missing message event`
Exampole code:
```
      const client = createClient({
        url: ''http://localhost:3000/graphql,
      })
      client.subscribe(
        { query: 'subscription {clock}' },
        {
          next: () => {
            console.log('yep')
          },
          error: (e) => {
            console.log('nope', e) <---- I get the error here
          },
          complete: () => {
            console.log('done')
          },
        }
      )
```

I would really appreciate any help

Thank you

**Comment by enisdenjo** [comment, marked as answer]
Sadly, [`graphql-helix`](https://github.com/contra/graphql-helix) does not implement the [GraphQL over Server-Sent Events Protocol](https://github.com/enisdenjo/graphql-sse/blob/master/PROTOCOL.md) that `graphql-sse` leverages; it uses its own protocol instead. Consider opening an issue over there.

**Reply by blackpr** [reply]
thank you
